codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java

```diff
@@ -144,20 +145,27 @@ public abstract class HttpObjectDecoder extends ByteToMessageDecoder {
     private final boolean allowPartialChunks;
     protected final boolean validateHeaders;
     private final boolean allowDuplicateContentLengths;
-    private final ByteBuf parserScratchBuffer;
     private final HeaderParser headerParser;
     private final LineParser lineParser;
-
+    private final AtomicBoolean resetRequested = new AtomicBoolean();
+    private final int initialBufferSize;
+    private ByteBuf parserScratchBuffer;
     private HttpMessage message;
     private long chunkSize;
     private long contentLength = Long.MIN_VALUE;
-    private final AtomicBoolean resetRequested = new AtomicBoolean();
 
     // These will be updated by splitHeader(...)
     private AsciiString name;
     private String value;
     private LastHttpContent trailer;
 
+    @Override
+    public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
+        // We depend on the fact that this buffer is a heap buffer.
+        parserScratchBuffer = ctx.alloc().heapBuffer(initialBufferSize);
```

Comment:
I wasn't sure to make it pooled, for good reasons...
Usually the life cycle of the handler is very long and we are basically pinning memory out of TLABs of the pooled allocator, likely...
The second reason is that I try hard to avoid heap arenas, because limit the heap capacity of the application which uses Netty.